app/src/main/java/org/sagebionetworks/research/mpower/tracking/view_model/TrackingTaskViewModel.java

```diff
@@ -29,63 +36,154 @@
         this.stepView = stepView;
-        this.availableElements = new MutableLiveData<>();
-        this.availableElements.setValue(stepView.getSelectionItems());
-        this.activeElements = new MutableLiveData<>();
-        this.activeElements.setValue(new HashSet<>());
-        this.selectionMade = Transformations.map(this.activeElements, (elements) -> !elements.isEmpty());
-        this.unconfiguredElements = Transformations.map(this.activeElements, (elements) -> {
-            Set<ConfigType> result = new HashSet<>();
-            for (ConfigType config : elements) {
-                if (!config.isConfigured()) {
-                    result.add(config);
-                }
+        availableElements = new MutableLiveData<>();
+        activeElementsById = new MutableLiveData<>();
+        loggedElementsById = new MutableLiveData<>();
+        availableElements.setValue(stepView.getSelectionItems());
+        trackingItemsById = getTrackingItemsById(availableElements.getValue());
+        loggedElementsById.setValue(new HashMap<>());
+        // initialize the active elements to contain either the user's previous selection or nothing depending on if
+        // we have a previous logging collection.
+        if (previousLoggingCollection == null) {
+            activeElementsById.setValue(new HashMap<>());
+        } else {
+            Map<String, ConfigType> activeElements = new HashMap<>();
+            for (LogType log : previousLoggingCollection.getItems()) {
+                String identifier = log.getIdentifier();
+                activeElements.put(identifier, instantiateConfigFromSelection(trackingItemsById.get(identifier)));
             }
 
-            return result;
-        });
+            activeElementsById.setValue(activeElements);
+        }
+    }
 
-        this.loggedElements = new MutableLiveData<>();
-        this.loggedElements.setValue(new HashSet<>());
+    protected static Map<String, TrackingItem> getTrackingItemsById(
+            Map<TrackingSection, Set<TrackingItem>> availableElements) {
+        Map<String, TrackingItem> result = new HashMap<>();
+        for (Set<TrackingItem> itemSet : availableElements.values()) {
+            for (TrackingItem item : itemSet) {
+                result.put(item.getIdentifier(), item);
+            }
+        }
+
+        return result;
     }
 
-    // region Selection
     /**
-     * Returns a LiveData which is true whenever the user has selected at least on item, and false otherwise.
+     * Sets the timestamp the task started at to the given instant.
      *
-     * @return a LiveData which is true whenever the user has selected at least on item and false otherwise.
+     * @param startDate
+     *         The instant to use as the start date for task.
      */
-    public LiveData<Boolean> getSelectionMade() {
-        return this.selectionMade;
+    public void setTaskStartDate(@Nullable Instant startDate) {
+        this.startDate = startDate;
```

Comment:
Shouldn't start be when the view model is created and the end be when you build the final logging collection?  Having the fragment set this will probably create unexpected dates when it is created/destroyed.